library/Validator.php

```diff
@@ -182,51 +184,42 @@ public function check($input): void
     }
 
     /**
+     * Creates a new instance Validator with a rule that was called on the static method.
+     *
      * @param string $ruleName
-     * @param array  $arguments
+     * @param array $arguments
      *
-     * @return Validator
+     * @throws ComponentException
+     *
+     * @return self
      */
-    public static function __callStatic($ruleName, $arguments)
+    public static function __callStatic(string $ruleName, array $arguments): self
     {
-        return (new static())->__call($ruleName, $arguments);
+        return self::create()->__call($ruleName, $arguments);
     }
 
     /**
-     * @param mixed $ruleSpec
+     * Create a new rule by the name of the method and adds the rule to the chain.
+     *
+     * @param string $ruleName
      * @param array $arguments
      *
-     * @return Validatable
-     */
-    public static function buildRule($ruleSpec, $arguments = [])
-    {
-        try {
-            return Factory::getDefaultInstance()->rule($ruleSpec, $arguments);
-        } catch (\Exception $exception) {
-            throw new ComponentException($exception->getMessage(), $exception->getCode(), $exception);
-        }
-    }
-
-    /**
-     * @param string $method
-     * @param array  $arguments
+     * @throws ComponentException
      *
      * @return self
      */
-    public function __call($method, $arguments)
+    public function __call(string $ruleName, array $arguments): self
     {
-        return $this->addRule(static::buildRule($method, $arguments));
+        return $this->addRule(Factory::getDefaultInstance()->rule($ruleName, $arguments));
     }
 
     /**
      * Create instance validator.
      *
      * @return Validator
      */
-    public static function create()
+    public static function create(): self
     {
-        $ref = new ReflectionClass(__CLASS__);
-
-        return $ref->newInstanceArgs(func_get_args());
```

Comment:
I wonder why we had this `func_get_args()`, for nothing? 🤔

Reply:
Was to allow `v::create(new Equals('somehing'))`